protocol/operation.re

```diff
@@ -38,33 +39,30 @@ module Side_chain = {
     (hash, verify);
   };
 
-  let make = (~nonce, ~block_height, ~source, ~amount, ~kind) => {
+  let verify =
+      (~hash, ~signature, ~nonce, ~block_height, ~source, ~amount, ~kind) => {
+    let.ok () =
+      verify(~hash, ~nonce, ~block_height, ~source, ~amount, ~kind)
+        ? Ok() : Error("Side operation invalid hash");
+    let.ok () =
+      Signature.verify(~signature, hash)
+      && Wallet.pubkey_matches_wallet(
+           Signature.public_key(signature),
+           source,
+         )
+        ? Ok() : Error("Side operation invalid signature");
```

Comment:
Should these be string errors or would variants be better?

Reply:
variants would change the type and bind would fail as one would be `result('a, string)` and the other `result('b, some_other_type)`

Reply:
resolving this to merge